### PR TITLE
Set authenticated user to print it in access logs

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
A server admin might want to track the usage of certain token or a user.

Modifications:
- Set `RequestLog.authenticatedUser()` to print it out via access logs.
- Bring back the log dependencies carelessly removed in #745

Result:
- You can track the usage of a token from access logs.

### CODE REVIEW

File: dist/build.gradle [hunk 1]
```diff
@@ -26,6 +26,16 @@ ext {
     cliDownloadDir = "${gradle.gradleUserHomeDir}/caches/centraldogma-go/cli"
 }
 
+
+dependencies {
+    // Logging
+    runtimeOnly libs.logback
+    runtimeOnly libs.armeria.logback
```
Comment (jrhee17) [inline dist/build.gradle:33]: Question) Is `libs.armeria.logback` actually used in this project?
Would it make sense to add this runtime jar if a project actually uses it?

File: server/src/test/resources/com/linecorp/centraldogma/server/metadata/testLoginStringifier.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  ~ License for the specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<configuration>
+  <appender name="NOP" class="ch.qos.logback.core.helpers.NOPAppender" />
+  <appender name="RCEA" class="com.linecorp.armeria.common.logback.RequestContextExportingAppender">
+    <appender-ref ref="NOP" />
+    <export>
+      attrs.app_id:com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil#CURRENT_USER:com.linecorp.centraldogma.server.metadata.LoginStringifier
```
Comment (jrhee17) [inline server/src/test/resources/com/linecorp/centraldogma/server/metadata/testLoginStringifier.xml:22]:
```suggestion
      attrs.user_id:com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil#CURRENT_USER:com.linecorp.centraldogma.server.metadata.LoginStringifier
```

File: server/src/test/java/com/linecorp/centraldogma/server/metadata/LoginStringifierTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        final ListAppender<ILoggingEvent> la = new ListAppender<>();
+        rcea.addAppender(la);
+        rcea.start();
+        la.start();
+        final ServiceRequestContext ctx = contextWithToken(sb -> {});
+        try (SafeCloseable ignored = ctx.push()) {
+            logger.trace("log");
+            assertThat(la.list.remove(0).getMDCPropertyMap()).containsExactly(
+                    Maps.immutableEntry("attrs.app_id", "foo"));
```
Comment (jrhee17) [inline server/src/test/java/com/linecorp/centraldogma/server/metadata/LoginStringifierTest.java:89]: nit; can use assertj's `entry`
```suggestion
                    entry("attrs.app_id", "foo"));
```

**Review by jrhee17 — COMMENTED** [review]
Looks good overall! Left a minor question 🙇

**Comment by minwoox** [comment]
Oops I should've told you that this PR will be changed after https://github.com/line/armeria/pull/4634 is merged. Sorry about it. 😓

**Comment by minwoox** [comment]
Let me update this after Armeria 1.22.0 is released.

**Comment by jrhee17** [comment]
Please go ahead and update this PR 🙏

File: dist/build.gradle (hunk 1 above)
Comment (minwoox) [reply dist/build.gradle]: It's used with the previous implementation which uses Stringifier. But now we don't need this, so I removed it. 😉

**Comment by minwoox** [comment]
> Please go ahead and update this PR 🙏

It's done. 😄

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java [hunk 4]
```diff
@@ -65,12 +65,13 @@ public CompletionStage<Boolean> authorize(ServiceRequestContext ctx, HttpRequest
...
                            if (appToken != null && appToken.isActive()) {
-                               final StringBuilder login = new StringBuilder(appToken.appId());
+                               final String appId = appToken.appId();
+                               final StringBuilder login = new StringBuilder(appId);
                                final SocketAddress ra = ctx.remoteAddress();
                                if (ra instanceof InetSocketAddress) {
                                    login.append('@').append(((InetSocketAddress) ra).getHostString());
                                }
-
+                               ctx.logBuilder().authenticatedUser(appId);
```
Comment (jrhee17) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java:74]: Question, should we also include `login` information?

**Review by jrhee17 — APPROVED** [review]
Looks good to me 👍 🙇 🚀

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java (hunk 4 above)
Comment (ikhoon) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java:74]: How about adding a prefix to easily distinguish `appId` and `username`?
`user/{username}`, `app/{appId}`?
```suggestion
                               ctx.logBuilder().authenticatedUser("app/" + appId);
```

File: server/src/main/java/com/linecorp/centraldogma/server/internal/admin/auth/SessionTokenAuthorizer.java [hunk 5]
```diff
@@ -62,6 +62,7 @@ public CompletionStage<Boolean> authorize(ServiceRequestContext ctx, HttpRequest
                                  final List<String> roles = administrators.contains(username) ? LEVEL_ADMIN
                                                                                               : LEVEL_USER;
                                  final User user = new User(username, roles);
+                                 ctx.logBuilder().authenticatedUser(username);
```
Comment (ikhoon) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/admin/auth/SessionTokenAuthorizer.java:65]:
```suggestion
                                 ctx.logBuilder().authenticatedUser("user/" + username);
```

**Review by ikhoon — COMMENTED** [review]
Left a minor suggestion.

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java (hunk 4 above)
Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/ApplicationTokenAuthorizer.java]: That's a good suggestion!

**Review by ikhoon — APPROVED** [review]
Good job!

**Review by trustin — APPROVED** [review]

**Pull request merged**